src/main/java/thestonedturtle/mahoganyhomes/MahoganyHomesPlugin.java

```diff
@@ -674,4 +706,35 @@ private void refreshTeleportItem(final WorldPoint playerPos)
 			teleportItem = null;
 		}
 	}
+
+	void updateResourcesInInventory()
```

Comment:
Due to the other requested changes we need to call this whenever a new contract is received.

Also, ensure this gets called when enabling the plugin if they have an active contract. This may need to be done inside `onStartUp` but I believe calling it in the above scenario should work.

Reply:
> Due to the other requested changes we need to call this whenever a new contract is received.

Done in checkForContractTierDialog()


> Also, ensure this gets called when enabling the plugin if they have an active contract. This may need to be done inside `onStartUp` but I believe calling it in the above scenario should work.

I actually found calling this at the end of loadFromConfig() to be more ideal, as it additionally covers my other scenarios.